### ISSUE TITLE
Button to increase/decrease page size

### ISSUE THREAD

**Issue by leonardoxopp** [xournalpp/xournalpp#2121]
Although it is easy to change paper size, I would like to have buttons in the toolbar to increase/decrease page size.

The attached image describes an idea.

![button](https://user-images.githubusercontent.com/68242604/87859879-0d1e3f80-c90f-11ea-849b-d5e278abf3d1.png)

Kind regards,
Leonardo

**Comment by rolandlo** [comment]
I do not think that would be widely used. Why would you change paper size often?

**Comment by leonardoxopp** [comment]
I'm using Xournal++ to create class notes. Sometimes in a particular page I just need a "few more paper" and I need to create a new one just to write one or two lines.

Of course this is not a big deal, but, at least for me, a shortcut to increase paper size would be helpful.

Anyway, I agree with you that "priority: low" is correct. There are a lot of other things to be done first.

Kind regards,
Leonardo

**Comment by rolandlo** [comment]
Ok, I see why it could be useful. So maybe that would be a good case for a plugin.

**Comment by leonardoxopp** [comment]
I was reading the plugins that comes with xournalpp's source code in order to write a simple plugin to just increase current page height, lets say, by 3.0 cm. However I could not find which variable controls "CurrentPageHeight".

Any clues?

Kind regards,
Leonardo

**Comment by rolandlo** [comment]
It is currently not possible without extending the lua plugin interface. 
In luapi_application.h there is already an applib function that changes the page background. You would have to write a similar applib function that changes the page size. Use plugin->getControl()->getDocument() to get a handle on the document. From there you have the setPageSize and the getPageSize functions. Then you can use your newly created applib function within the plugin.

**Comment by Rmano** [comment]
> I'm using Xournal++ to create class notes. Sometimes in a particular page I just need a "few more paper" and I need to create a new one just to write one or two lines.

I would heartily advise against doing that. If you then convert it to PDF for archival reasons, it will become a small nightmare both in viewing (different zooms while going back and forward) and for printing (same). 

I think that you stick to one paper size, or use an infinite-length paper "roll" (and forget about ever printing them). 

Yep, I know that the "infinite paper" is not available too ;-)

**Comment by leonardoxopp** [comment]
@rolandlo I will take a look, thanks.

@Rmano thanks for this advice. I will explain my particular case and why your concern won't be a problem. 
Also, the "infinite paper" would be an amazing feature.

First: I begin with a paper size of 27.2 x 15.3 cm (after try and error, this size fits well), and prepare some notes (I'm using a pen tablet) with theory (without examples - math lessons) and I create the PDF just for security/backup. Here the "increase paper size" is not necessary .

Second: I use ffmpeg+mplayer to record video-lessons. In this step I explain the content and write the examples.
Sometimes, just to finish a particular example, I would like to increase paper size just a bit (3 cm or 6 cm) to avoid create a new page (because of that "jumping screen" problem already reported).
Since here I'm only interested in the video that I'm recording, the PDF with examples is not needed.

Anyway, this is not a big issue. I will try to create the plugin and if I can't do, no problem.

**Comment by rolandlo** [comment]
After PR #2406 is merged, you will have the plugin API command `app.setPageSize` function available. For instance `app.setPageSize(0,14.17*6, true)` will add 3cm to the page height. If you have a keyboard available you can activate the plugin via keyboard accelerator.
